src/adapter.ts

```diff
@@ -696,7 +697,16 @@ export class USBAdapter extends EventEmitter implements Adapter {
             const type = this.controlTransferParamsToType(setup, LIBUSB_ENDPOINT_IN);
 
             device.controlTransfer(type, setup.request, setup.value, setup.index, length, (error, buffer) => {
-                if (error) return reject(error);
+                if (error) {
+                    if (error.errno == LIBUSB_TRANSFER_STALL) {
+                        return resolve({
+                            status: "stall"
```

Comment:
Is `bytesWritten` needed here?

Reply:
No, it's a transfer in, so it would be `data`, not `bytesWritten`. It's not clear to me if a partial transfer could have occurred and there may be data available. The order of the steps in the spec make that possible, but I'm not sure if `libusb` provides that. I don't have a reproduce case for a transfer in, only out, so I can't test myself.

From the spec:

```
9. If data was received from device create a new ArrayBuffer containing the first bytesTransferred bytes of buffer and set result.data to a new DataView constructed over it.

10. If device responded by stalling the default control pipe set result.status to "stall".
```

So the open question is whether `buffer` is an object and has any data if a stall happens on transfer in.

Reply:
I see, happy with it as is.